hudi-client/hudi-spark-client/src/main/scala/org/apache/hudi/HoodieInternalRowUtils.scala

```diff
@@ -267,8 +277,23 @@ object HoodieInternalRowUtils {
     }
   }
 
+  def broadcastCompressedSchema(schemas: List[StructType], sc: SparkContext): Unit = {
```

Comment:
Let's avoid over-complicating the implementation unnecessarily -- as we've discussed before we don't need to share the caches b/w `HoodieSparkRecordSerializer` and `HoodieInternalRowUtils` they are fine to have separate caches of their own.